bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/NetUtil.java

```diff
@@ -55,9 +60,53 @@ public static String getLocalIpv4HostAddress() {
             }
             return hostAddress;
         } catch (SocketException ex) {
-            LOGGER.error("Could not retrieve network interface: " + ex.getMessage(), ex);
+            LOGGER.error("Could not retrieve network interface: {}", ex.getMessage(), ex);
             return null;
         }
     }
 
+    /**
+     * Get all broadcast addresses on the current host
+     *
+     * @return list of broadcast addresses
+     */
+    public static List<String> getAllBroadcastAddresses() {
+        try {
+            List<String> broadcastAddresses = new ArrayList<String>();
+            final Enumeration<NetworkInterface> networkInterfaces = NetworkInterface.getNetworkInterfaces();
+            while (networkInterfaces.hasMoreElements()) {
+                final NetworkInterface networkInterface = networkInterfaces.nextElement();
+                final List<InterfaceAddress> interfaceAddresses = networkInterface.getInterfaceAddresses();
+                for (InterfaceAddress interfaceAddress : interfaceAddresses) {
+                    final InetAddress addr = interfaceAddress.getAddress();
+                    if (!addr.isLinkLocalAddress() && !addr.isLoopbackAddress() && addr instanceof Inet4Address) {
+                        broadcastAddresses.add(interfaceAddress.getBroadcast().getHostAddress());
+                    }
+                }
+            }
+            return broadcastAddresses;
+        } catch (SocketException ex) {
+            LOGGER.error("Could not find broadcast address: {}", ex.getMessage(), ex);
+            return null;
+        }
+    }
+
+    /**
+     * Get the first candidate for a broadcast address
+     *
+     * @return broadcast address
+     */
+    public static String getBroadcastAddress() {
+        String broadcastAddress = null;
+        final List<String> broadcastAddresses = getAllBroadcastAddresses();
+        for (String current : broadcastAddresses) {
+            if (broadcastAddress != null) {
+                LOGGER.warn("Found multiple broadcast addresses - ignoring {}", broadcastAddress);
```

Comment:
The JavaDoc state "get the first candidate", so I don't see any reason to throw a warning if there are multiple candidates. The function should return the first one and not check if there is more then one, or should it?